test/distributed/checkpoint/test_list_stored_state_dict.py

```diff
@@ -0,0 +1,63 @@
+# Owner(s): ["oncall: distributed checkpointing"]
+
+import io
+
+import torch
+import torch.distributed.checkpoint as dist_cp
+from torch.distributed.checkpoint.metadata import (
+    BytesStorageMetadata,
+    TensorStorageMetadata,
+)
+from torch.distributed.device_mesh import init_device_mesh
+from torch.distributed.tensor import distribute_tensor, Replicate, Shard
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    skip_if_lt_x_gpu,
+    with_comms,
+)
+from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
+
+
+CHECKPOINT_DIR = "checkpoint"
+
+
+class TestListStateDict(DTensorTestBase):
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    @with_temp_dir
+    def test_list_stored_sd_metadata(self) -> None:
+        global_tensor = torch.arange(16, dtype=torch.float).view(4, 4)
+        mesh_shape = (2, self.world_size // 2)
+        mesh_2d = init_device_mesh(self.device_type, mesh_shape)
+        # Save 1) DTensor 2) Blob
+        dtensor = distribute_tensor(
+            global_tensor,
+            mesh_2d,
+            placements=[Shard(0), Replicate()],
+        )
+        state_dict_to_save = {
+            "distributed_weight": dtensor,
+            "bytes_data": io.BytesIO(b"TrainingEpoch:4"),
+        }
+
+        dist_cp.save(
+            state_dict=state_dict_to_save,
+            storage_writer=dist_cp.FileSystemWriter(path=CHECKPOINT_DIR),
+            planner=dist_cp.DefaultSavePlanner(),
+        )
```

Comment:
Your git porcelain is failing since your test run didn't properly clean up its files after things run. We do this to ensure developers local directories don't fill up with files that could potentially be staged for committing.

You need to ensure this file gets deleted, or ideally use a tempfile.

See https://docs.python.org/3/library/tempfile.html